File: testdata/minimal-testbeds/spring/src/main/java/com/example/demo/OwnerController.java
package com.example.demo;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class OwnerController {
  private final PetService pets;

  public OwnerController(PetService pets) {
    this.pets = pets;
  }

  @GetMapping("/owners/{name}")
  public String greet(@PathVariable String name) {
    return this.pets.greet(name);
  }
}

File: testdata/minimal-testbeds/kotlin/src/main/kotlin/com/example/demo/OwnerController.kt
package com.example.demo

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class OwnerController(
    private val pets: PetService
) {
    @GetMapping("/owners/{name}")
    fun greet(@PathVariable name: String): String {
        return pets.greet(name)
    }
}
